### PR TITLE
fixed some codacy issues

### PR DESCRIPTION
**Pull request by jstucke**
https://app.codacy.com/app/johannes.vom.dorp/FACT_core/issues

### CODE REVIEW

File: src/compile_yara_signatures.py [hunk 1]
```diff
@@ -20,7 +19,9 @@
 
 import os
 import subprocess
+from shlex import split
```
Comment (dorpvom) [inline src/compile_yara_signatures.py]: (see below) Then you also don't need this import

File: src/helperFunctions/yara_binary_search.py [hunk 2]
```diff
@@ -24,11 +25,8 @@ def _execute_yara_search(self, rule_file_path, target_path=None):
         :param rule_file_path: file path to yara rule file
         :return: output from yara scan
         '''
-        return check_output(
-            'yara -r {} {}'.format(rule_file_path, self.db_path if target_path is None else target_path),
-            shell=True,
-            stderr=STDOUT
-        )
+        command = 'yara -r {} {}'.format(rule_file_path, self.db_path if target_path is None else target_path)
+        return check_output(split(command), stderr=STDOUT)
```
Comment (dorpvom) [inline src/helperFunctions/yara_binary_search.py]: See above: Change to common_helper_process

File: src/start_fact.py [hunk 3]
```diff
@@ -64,7 +68,7 @@ def _terminate_process(process):
             process.kill()
 
 
-def shutdown(signum, frame):
+def shutdown(*_):
```
Comment (dorpvom) [inline src/start_fact.py]: Consider adding `**__` to avoid running into runtime issues

File: src/start_fact_backend.py [hunk 4]
```diff
@@ -34,7 +34,7 @@
 PROGRAM_DESCRIPTION = 'Firmware Analysis and Compare Tool (FACT) Backend'
 
 
-def shutdown(signum, frame):
+def shutdown(*_):
```
Comment (dorpvom) [inline src/start_fact_backend.py]: see above

File: src/start_fact_db.py [hunk 5]
```diff
@@ -30,7 +30,7 @@
 PROGRAM_DESCRIPTION = 'Firmware Analysis and Compare Tool (FACT) DB-Service'
 
 
-def shutdown(signum, frame):
+def shutdown(*_):
```
Comment (dorpvom) [inline src/start_fact_db.py]: see above

File: src/start_fact_frontend.py [hunk 6]
```diff
@@ -54,19 +55,20 @@ def _shutdown_uwsgi_server(process):
...
-    return p
+    command = 'uwsgi --ini  {}/uwsgi_config.ini{}'.format(get_config_dir(), config_parameter)
+    process = Popen(split(command), cwd=get_src_dir())
+    return process
 
 
 def start_docker():
     command = 'docker-compose -f {}/install/radare/docker-compose.yml up -d'.format(get_src_dir())
-    with Popen(command, shell=True, stdout=PIPE, stderr=PIPE) as docker_process:
+    with Popen(split(command), stdout=PIPE, stderr=PIPE) as docker_process:
```
Comment (dorpvom) [inline src/start_fact_frontend.py]: see above: can this be done with common_helper_process?

Also line 71

File: src/statistic/update.py [hunk 7]
```diff
@@ -303,11 +303,15 @@ def _shorten_architecture_string(s):
             # short string (without bitness and endianness but with ' (M)' at the end)
             return s[:-4]
 
-    def _get_ip_stats(self):
-        stats = {}
-        stats['ips_v4'] = self._get_objects_and_count_of_occurrence_firmware_and_file_db('$processed_analysis.ip_and_uri_finder.ips_v4', unwind=True, sumup_function=sum_up_nested_lists)
-        stats['ips_v6'] = self._get_objects_and_count_of_occurrence_firmware_and_file_db('$processed_analysis.ip_and_uri_finder.ips_v6', unwind=True, sumup_function=sum_up_nested_lists)
-        stats['uris'] = self._get_objects_and_count_of_occurrence_firmware_and_file_db('$processed_analysis.ip_and_uri_finder.uris', unwind=True)
+    def get_ip_stats(self):
```
Comment (dorpvom) [inline src/statistic/update.py]: Ehm, why not one line like in 221 once your at it?

File: src/storage/MongoMgr.py [hunk 8]
```diff
@@ -62,10 +63,11 @@ def check_file_and_directory_existence_and_permissions(self):
...
             logging.info('stop local mongo database')
-            with Popen('mongo --eval "db.shutdownServer()" {}:{}/admin --username {} --password \'{}\''.format(
-                    self.config['data_storage']['mongo_server'], self.config['data_storage']['mongo_port'],
-                    self.config['data_storage']['db_admin_user'], self.config['data_storage']['db_admin_pw']),
-                    shell=True, stdout=PIPE, stderr=STDOUT) as pl:
+            command = 'mongo --eval "db.shutdownServer()" {}:{}/admin --username {} --password "{}"'.format(
+                self.config['data_storage']['mongo_server'], self.config['data_storage']['mongo_port'],
+                self.config['data_storage']['db_admin_user'], self.config['data_storage']['db_admin_pw']
+            )
+            with Popen(split(command), stdout=PIPE, stderr=STDOUT) as pl:
```
Comment (dorpvom) [inline src/storage/MongoMgr.py]: Maybe common_helper_process?

File: src/web_interface/filter.py [hunk 9]
```diff
@@ -337,7 +341,9 @@ def user_has_role(current_user, role):
     return current_user.is_authenticated and user_has_privilege(current_user, role)
 
 
-def sort_roles_by_number_of_privileges(roles, privileges=PRIVILEGES):
+def sort_roles_by_number_of_privileges(roles, privileges=None):
+    if privileges is None:
```
Comment (dorpvom) [inline src/web_interface/filter.py]: You can replace this and all other `if ... is None:` with `<var> = <var> if <var> else <default>`

File: src/start_fact.py (hunk 3 above)
Comment (jstucke) [reply src/start_fact.py]: this should not be an issue since it was like this in `start_fact_frontend.py` for several months without problems

**Pull request merged**